### PR TITLE
[DO NOT MERGE] cmake: Migrate CI scripts to CMake-based build system -- WIP

### PR DESCRIPTION
**Pull request by hebasto**
This PR goal is to facilitate testing CI scripts migration to the CMake.

### CODE REVIEW

**Comment by hebasto** [comment]
The recent push uses approach from the https://github.com/hebasto/bitcoin/pull/157.

**Comment by hebasto** [comment]
Rebased and updated with usage of https://github.com/hebasto/bitcoin/pull/184.

File: ci/test/00_setup_env_native_fuzz_with_msan.sh [hunk 1]
```diff
@@ -18,7 +18,7 @@ export PACKAGES="ninja-build"
 export DEP_OPTS="DEBUG=1 NO_BDB=1 NO_QT=1 CC=clang CXX=clang++ CFLAGS='${MSAN_FLAGS}' CXXFLAGS='${MSAN_AND_LIBCXX_FLAGS}'"
 export GOAL="install"
 # _FORTIFY_SOURCE is not compatible with MSAN.
-export BITCOIN_CONFIG="--enable-fuzz --with-sanitizers=fuzzer,memory CPPFLAGS='-DBOOST_MULTI_INDEX_ENABLE_SAFE_MODE -U_FORTIFY_SOURCE'"
+export BITCOIN_CONFIG="-DFUZZ=ON -DSANITIZERS=fuzzer,memory -DHARDENING=OFF -DAPPEND_CPPFLAGS='-DBOOST_MULTI_INDEX_ENABLE_SAFE_MODE -U_FORTIFY_SOURCE'"
```
Comment (fanquake) [inline ci/test/00_setup_env_native_fuzz_with_msan.sh:21]: Should remove `-DHARDENING=OFF` here.

Comment (hebasto) [reply ci/test/00_setup_env_native_fuzz_with_msan.sh]: Thanks! Removed.

**Comment by hebasto** [comment]
@maflcko Would you mind having a look into this, please?

People who works on CMake migration are highly anticipating your opinion on use cases for [`APPEND_{CPP,C,CXX,LD}FLAGS`](https://github.com/hebasto/bitcoin/pull/184) in the CI scripts.

cc @theuni

**Comment by hebasto** [comment]
This branch has been rebased on top of the https://github.com/hebasto/bitcoin/pull/202.

I've noticed two issues so far:
1. `Cannot open mapping file '/ci_container_base/ci/scratch/build-/contrib/devtools/iwyu/bitcoin.core.imp': No such file or directory.`
2. Missed disabling `EXPORT_COMPILE_COMMANDS` property for targets in the `crc32c` subtree.

**Comment by hebasto** [comment]
> I've noticed two issues in the "tidy" job so far:
> 
>    1. `Cannot open mapping file '/ci_container_base/ci/scratch/build-/contrib/devtools/iwyu/bitcoin.core.imp': No such file or directory.`
> 
>    2. Missed disabling `EXPORT_COMPILE_COMMANDS` property for targets in the `crc32c` subtree.

The recent push includes:
- commits from https://github.com/hebasto/bitcoin/pull/205, which fixes the point 1
- commits from https://github.com/hebasto/bitcoin/pull/206, which fixes the point 2

**Comment by hebasto** [comment]
> I've noticed two issues in the "tidy" job so far:
> 
>    1. `Cannot open mapping file '/ci_container_base/ci/scratch/build-/contrib/devtools/iwyu/bitcoin.core.imp': No such file or directory.`

The recent push includes commits from https://github.com/hebasto/bitcoin/pull/205, which fixes the point 1.

>    2. Missed disabling `EXPORT_COMPILE_COMMANDS` property for targets in the `crc32c` subtree.

The point 2 has been fixed already.

**Comment by hebasto** [comment]
Rebased. Same as https://github.com/hebasto/bitcoin/pull/220.

**Comment by hebasto** [comment]
Closing. See https://github.com/bitcoin/bitcoin/pull/30454.

**Comment by hebasto** [comment]
Temporarily reopening this PR to demonstrate the correctness of the https://github.com/hebasto/bitcoin/pull/273.

**Pull request closed without merging**